cmd/neofs-node/cache.go

```diff
@@ -278,12 +293,61 @@ func (s *ttlContainerLister) List(id *user.ID) ([]cid.ID, error) {
 		return nil, err
 	}
 
-	return val.([]cid.ID), nil
+	// panic on typecast below is OK since developer must be careful,
+	// runtime can do nothing with wrong type occurrence
+	item := val.(*cacheItemContainerList)
+
+	item.mtx.RLock()
+	res := make([]cid.ID, len(item.list))
+	copy(res, item.list)
+	item.mtx.RUnlock()
+
+	return res, nil
 }
 
-// InvalidateContainerList removes cached list of container IDs.
-func (s *ttlContainerLister) InvalidateContainerList(id user.ID) {
-	(*ttlNetCache)(s).remove(id.EncodeToString())
+// updates cached list of owner's containers: cnr is added if flag is true, otherwise it's removed.
+// Concurrent calls can lead to some races:
+//   - two parallel additions to missing owner's cache can lead to only one container to be cached
```

Comment:
I don't wanna solve it via global mutex since it'll slow down processing in multi-user environment, and in practice parallel creations for the single user aren't really expected - separate notifications. So missing ID of the created container in listing for `TTL` period in rare cases is better than global slowdown IMO.